protocol/x/clob/memclob/orderbook.go

```diff
@@ -1,56 +1,80 @@
 package memclob
 
 import (
-	errorsmod "cosmossdk.io/errors"
 	"fmt"
 	"math"
 
-	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
 	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
 	"github.com/pkg/errors"
 	"github.com/zyedidia/generic/list"
 )
 
-// memclobOpenOrders is a utility struct used for storing orders (within an orderbook) and order expirations.
-type memclobOpenOrders struct {
-	// Holds every `Orderbook` by ID of the CLOB.
-	orderbooksMap map[types.ClobPairId]*types.Orderbook
+// Orderbook holds the bids and asks for a specific product.
+type Orderbook struct {
+	// Defines the tick size of the orderbook by defining how many subticks
+	// are in one tick. That is, the subticks of any valid order must be a
+	// multiple of this value. Generally this value should start `>= 100` to
+	// allow room for decreasing it. This field is stored in state as part of a
+	// `ClobPair`, but must be made available to the in-memory `Orderbook` in
+	// order to efficiently remove orders from the orderbook. See the `removeOrder`
+	// implementation for more information.
+	SubticksPerTick types.SubticksPerTick
+	// Map of price level (in subticks) to buy orders contained at that level.
+	Bids map[types.Subticks]*types.Level
+	// Map of price level (in subticks) to sell orders contained at that level.
+	Asks map[types.Subticks]*types.Level
+	// The highest bid on this orderbook, in subticks. 0 if no bids exist.
+	BestBid types.Subticks
+	// The lowest ask on this orderbook, in subticks. math.MaxUint64 if no asks exist.
+	BestAsk types.Subticks
+	// Contains all open orders on this CLOB for a given subaccount and side.
+	// Used for fetching open orders for the add to orderbook collateralization
+	// check for a subaccount.
+	SubaccountOpenClobOrders map[satypes.SubaccountId]map[types.Order_Side]map[types.OrderId]bool
+	// Minimum size of an order on the CLOB, in base quantums.
+	MinOrderBaseQuantums satypes.BaseQuantums
+	// Contains all open reduce-only orders on this CLOB from each subaccount. Used for tracking
+	// which open reduce-only orders should be canceled when a position changes sides.
+	SubaccountOpenReduceOnlyOrders map[satypes.SubaccountId]map[types.OrderId]bool
+	// TotalOpenOrders tracks the total number of open orders in an orderbook for observability purposes.
+	TotalOpenOrders uint
 	// Map from order IDs to their `LevelOrder` reference contained in the
 	// orderbook, necessary for O(1) order removal from the orderbook.
 	orderIdToLevelOrder map[types.OrderId]*types.LevelOrder
 	// Map from block number to a set of all orders that expire at this block
 	// (with each order keyed by `OrderId`). Necessary for O(1) order removal
 	// from the orderbook when expiring orders in the EndBlocker.
 	blockExpirationsForOrders map[uint32]map[types.OrderId]bool
+	// A map of all known canceled order IDs mapped to their expiry block.
```

Comment:
nit: Non-blocking, given that you've updated the orderbook struct to hold both information about orders and order cancelations, maybe renaming these 2 maps to reference the word `cancel` would make reading this code clearer?